prov/efa/src/efa_cntr.c

```diff
@@ -85,7 +95,7 @@ int efa_cntr_open(struct fid_domain *domain, struct fi_cntr_attr *attr,
 		return -FI_ENOMEM;
 
 	ret = ofi_cntr_init(&efa_prov, domain, attr, cntr,
-			    &ofi_cntr_progress, context);
```

Comment:
Was ofi_cntr_progress used else where?

Reply:
It's the only place in efa provider.

Reply:
It is out of scope of this PR, but I wonder if other provider use this function and need to make similar change.
I

Reply:
This locking is required after efa onboards the util SRX framework, which requires the owner provider acquire a domain level lock (the srx_ctx->lock here) outside the progress engine. Currently efa is the only owner provider that onboards the util SRX, so other providers shouldn't need this change